devlib/host.py

```diff
@@ -55,7 +56,10 @@ def pull(self, source, dest, timeout=None, as_root=False): # pylint: disable=unu
             for each_source in iglob(source):
                 shutil.copy(each_source, dest)
         else:
-            shutil.copy(source, dest)
+            if os.path.isdir(source):
+                copy_tree(source, dest)
```

Comment:
Why not use `shutil.copytree` here?

Reply:
`shutil.copytree` does not support copying into an existing directory structure whereas the implementation from `distutils` does, therefore to try and mimic the behaviour of other targets as closely as possible I opted for this implementation.

Reply:
OK that makes sense. It's worth leaving a comment to that effect in the code.

Reply:
Updated.